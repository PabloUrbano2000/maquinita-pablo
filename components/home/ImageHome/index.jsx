import React, { useEffect, useRef, useState } from 'react';
import Box from '@mui/material/Box';
import Image from 'next/image';

const ImageHome = ({ mediaQuerySm }) => {
  const container = useRef();
  const [width, setWidth] = useState(0);
  const [height, setHeight] = useState(0);

  useEffect(() => {
    if (container) {
      if (mediaQuerySm) {
        setWidth((container.current.clientWidth * 80) / 100);
      } else {
        setWidth(container.current.clientWidth);
      }
      setHeight(container.current.clientHeight);
    }
  }, [container]);

  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        height: '100%',
        width: '100%'
      }}
      ref={container}
    >
      <Image
        src={'/img/img_home.svg'}
        alt="img-slot-t"
        width={width}
        height={height}
      />
    </Box>
  );
};

export default ImageHome;
