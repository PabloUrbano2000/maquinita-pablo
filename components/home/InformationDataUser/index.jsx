import React from 'react';
import Button from '@mui/material/Button';
import Title from '../../utils/Title';
import SubTitle from '../../utils/SubTitle';
import Box from '@mui/material/Box';
// import { useRouter } from 'next/router';

const InformationDataUser = ({ data }) => {
  //   const router = useRouter();

  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        gap: 5
      }}
      pl={'10%'}
      pr={'10%'}
    >
      <div>
        <Title
          style={{
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'center',
            alignItems: 'center',
            fontSize: '2rem',
            marginBottom: 20,
            textAlign: 'center'
          }}
        >
          {data.informationDataUser ? data.informationDataUser.messageOne : ''}
        </Title>

        <SubTitle
          style={{ fontSize: '1.5rem', display: 'block', textAlign: 'center' }}
        >
          {data.informationDataUser ? data.informationDataUser.messageTwo : ''}
        </SubTitle>
      </div>
      <div>
        <Button
          variant="contained"
          style={{
            fontSize: '1.4rem',
            paddingLeft: 40,
            paddingRight: 40,
            borderRadius: 80
          }}
          onClick={() => console.log('redirigir')}
        >
          <div>
            <div>
              {data.informationDataUser
                ? data.informationDataUser.ButtonOne
                : ''}
            </div>
            <div style={{ fontSize: '1rem' }}>
              {data.informationDataUser
                ? data.informationDataUser.ButtonTwo
                : ''}
            </div>
          </div>
        </Button>
      </div>
    </Box>
  );
};

export default InformationDataUser;
