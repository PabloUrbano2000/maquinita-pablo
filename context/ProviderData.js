import React, { useState, createContext, useEffect } from 'react';

export const DATA = {
  information: {
    messageOne: '¡PRUEBA TU SUERTE HOY!',
    messageTwo: 'Si te salen los tres',
    messageThree: 'de Trome',
    messageFour: 'ganas el premiazo',
    messageFive: '¡AL INSTANTE!'
  },
  buttons: {
    main: '¡HAZ CLIC AQUÍ!',
    terms: 'TÉRMINOS Y CONDICIONES',
    more: 'CONOCE MÁS',
    participates: 'PARTICIPA'
  },
  knowMore: {
    messageOne: 'Tu puedes ser uno de los ganadores de',
    messageTwo: '4 premios de s/',
    messageThree: '200',
    messageFour: 'Preguntas frecuentes:',
    questions: [
      {
        question: '¿Cómo se si soy el ganador?',
        answer: `Si al girar la ruleta te aparecen las tres T de trome,
      <b>GANASTE!</b> Automáticamente te aparecerá el código
      ganador.`
      },
      {
        question: '¿Cuánto tiempo tengo que esperar por mi premio?',
        answer: `En un plazo de <b>60 días hábiles</b> nos estaremos
      contactando contigo para coordinar la entrega del premio.`
      },
      {
        question: '¿Si gané hoy día puedo ganar cualquier otro?',
        answer: `Si! En esta promoción puedes participar <b>todos los días.</b>`
      },
      {
        question: '¿Puedo compartir mi código ganador con algún familiar?',
        answer: `No, el código ganador es <b>personal</b> y está asociado a tu
        número de DNI`
      },
      {
        question: 'Perdí mi código, ¿cómo puedo recuperarlo?',
        answer: `<b>Escríbele a Fito al 977 944 966</b> y él podrá ayudarte a
        recuperar tú código.`
      }
    ]
  },
  game: {
    messageOne: 'Prueba tu suerte HOY y participa de',
    buttonJ: 'Jugar',
    buttonD: 'Detener'
  },
  informationDataUser: {
    messageOne: 'Ya casi terminas',
    messageTwo: 'Completa los datos y participa por el premiazo',
    ButtonOne: 'Clic Aquí',
    ButtonTwo: 'y completa tus datos'
  }
};

export const DataContext = createContext({
  systemData: '',
  setSystemData: () => {},
  auth: '',
  setAuth: () => {}
});

const ProviderData = ({ children }) => {
  const [systemData, setSystemData] = useState(DATA);
  const [auth, setAuth] = useState(null);

  useEffect(() => {
    window.tp.push([
      'init',
      function () {
        // esta función se ejecuta en cola, luego de que Piano se inicialice
        // console.log('hola', window.tp.pianoId.isUserValid());
        if (window.tpuser) {
          setAuth(window.tpuser);
        }
      }
    ]);
  }, []);

  const context = {
    systemData,
    setSystemData,
    auth,
    setAuth
  };

  return (
    <DataContext.Provider value={context}>{children}</DataContext.Provider>
  );
};

export default ProviderData;
