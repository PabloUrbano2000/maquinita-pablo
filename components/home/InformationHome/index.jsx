import React from 'react';
import Button from '@mui/material/Button';
import Title from '../../utils/Title';
import SubTitle from '../../utils/SubTitle';
import Checkbox from '@mui/material/Checkbox';
import Box from '@mui/material/Box';
import CloseIcon from '@mui/icons-material/Close';
import Image from 'next/image';
import { useRouter } from 'next/router';
import InformationDataUser from '../InformationDataUser';

const InformationHome = ({
  data,
  handleGame,
  fillInData,
  setShowDialog,
  setErrorMessage
}) => {
  const router = useRouter();

  return fillInData ? (
    <InformationDataUser data={data} />
  ) : (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        gap: 5
      }}
      pl={'10%'}
      pr={'10%'}
    >
      <div>
        <Title
          style={{
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'center',
            alignItems: 'center',
            fontSize: '2rem',
            marginBottom: 20,
            textAlign: 'center'
          }}
        >
          {data.information ? data.information.messageOne : ''}
        </Title>
        <div
          style={{
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'center',
            alignItems: 'center',
            textAlign: 'center'
          }}
        >
          <SubTitle style={{ fontSize: '1.5rem', display: 'block' }}>
            {data.information ? data.information.messageTwo : ''}{' '}
            <Image
              src={'/img/slot/TROME-CostumerJourney-Icono-Ta.png'}
              alt="img-slot-t"
              width={20}
              height={20}
            />{' '}
            {data.information ? data.information.messageThree : ''}{' '}
            <SubTitle style={{ fontSize: '1.5rem' }}>
              {data.information ? data.information.messageFour : ''}{' '}
              <Title style={{ fontSize: '2rem' }}>
                {data.information ? data.information.messageFive : ''}
              </Title>
            </SubTitle>
          </SubTitle>
        </div>
      </div>
      <div>
        <Button
          variant="contained"
          style={{
            fontSize: '1.4rem',
            paddingLeft: 40,
            paddingRight: 40,
            borderRadius: 80
          }}
          onClick={() => handleGame()}
        >
          {data.buttons ? data.buttons.main : ''}
        </Button>
      </div>
      <Box
        sx={{
          display: 'flex',
          justifyContent: 'row',
          gap: 2
        }}
      >
        <Checkbox
          sx={{
            marginTop: 0.8,
            width: 15,
            height: 15,
            color: 'black',
            backgroundColor: '#ffffff',
            boxShadow: 'none',
            '&:hover': { backgroundColor: '#ffffff' }
          }}
          checked={true}
          checkedIcon={<CloseIcon htmlColor="#000000" />}
        />
        <div>
          <SubTitle style={{ fontSize: '1.2rem', display: 'flex' }}>
            Acepto los{' '}
            <Button
              variant="contained"
              style={{ fontSize: '0.4rem', marginLeft: 10 }}
              onClick={() => {
                setErrorMessage('Estos son los terminos y condiciones');
                setShowDialog(true);
              }}
            >
              {data.buttons ? data.buttons.terms : ''}
            </Button>
          </SubTitle>
          <SubTitle style={{ fontSize: '1.2rem', display: 'block' }}>
            al participar de la promoción{' '}
            <b style={{ display: 'block' }}>La Máquina Ganadora</b>
          </SubTitle>
        </div>
      </Box>
    </Box>
  );
};

export default InformationHome;
