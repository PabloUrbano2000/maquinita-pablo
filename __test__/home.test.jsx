import { render } from '@testing-library/react';
import ImageHome from '../components/home/ImageHome';
import InformationHome from '../components/home/InformationHome';
import InformationDataUser from '../components/home/InformationDataUser';
import { DATA } from '../context/ProviderData';

it('render component ImageHome', () => {
  const { container } = render(<ImageHome />);
  expect(container).toMatchSnapshot();
});

it('render component InformationHome', () => {
  const { container } = render(
    <InformationHome
      data={DATA}
      handleGame={() => console.log('actions game')}
      fillInData={true}
      setShowDialog={() => console.log('set value')}
      setErrorMessage={() => console.log('set value')}
    />
  );
  expect(container).toMatchSnapshot();
});

it('render component InformationDataUser', () => {
  const { container } = render(<InformationDataUser data={DATA} />);
  expect(container).toMatchSnapshot();
});
