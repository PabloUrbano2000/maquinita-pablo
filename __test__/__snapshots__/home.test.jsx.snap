// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component ImageHome 1`] = `
<div>
  <div
    class="MuiBox-root css-1lgihvy"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%270%27%20height=%270%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="img-slot-t"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
</div>
`;

exports[`render component InformationDataUser 1`] = `
<div>
  <div
    class="MuiBox-root css-18z1sft"
  >
    <div>
      <span
        class="css-yzbrn7"
        style="display: flex; flex-direction: column; justify-content: center; align-items: center; font-size: 2rem; margin-bottom: 20px; text-align: center;"
      >
        Ya casi terminas
      </span>
      <span
        class="css-8rchdx"
        style="font-size: 1.5rem; display: block; text-align: center;"
      >
        Completa los datos y participa por el premiazo
      </span>
    </div>
    <div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        style="font-size: 1.4rem; padding-left: 40px; padding-right: 40px; border-radius: 80px;"
        tabindex="0"
        type="button"
      >
        <div>
          <div>
            Clic Aquí
          </div>
          <div
            style="font-size: 1rem;"
          >
            y completa tus datos
          </div>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`render component InformationHome 1`] = `
<div>
  <div
    class="MuiBox-root css-18z1sft"
  >
    <div>
      <span
        class="css-yzbrn7"
        style="display: flex; flex-direction: column; justify-content: center; align-items: center; font-size: 2rem; margin-bottom: 20px; text-align: center;"
      >
        Ya casi terminas
      </span>
      <span
        class="css-8rchdx"
        style="font-size: 1.5rem; display: block; text-align: center;"
      >
        Completa los datos y participa por el premiazo
      </span>
    </div>
    <div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        style="font-size: 1.4rem; padding-left: 40px; padding-right: 40px; border-radius: 80px;"
        tabindex="0"
        type="button"
      >
        <div>
          <div>
            Clic Aquí
          </div>
          <div
            style="font-size: 1rem;"
          >
            y completa tus datos
          </div>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
